index.d.ts

```diff
@@ -0,0 +1,325 @@
+// Type definitions for Chalk
+// Project: Chalk
+// Definitions by: Thomas Sauer <https://github.com/t-sauer>
+// Updated definitions by: George Gkasdrogkas <https://github.com/georgegkas>
+
+export const enum Level {
+	/**
+	 * All colors disabled.
+	 */
+	None = 0,
+
+	/**
+	 * Basic 16 colors support.
+	 */
+	Basic = 1,
+
+	/**
+	 * ANSI 256 colors support.
+	 */
+	Ansi256 = 2,
+
+	/**
+	 * Truecolor 16 million colors support.
+	 */
+	TrueColor = 3
+}
+
+export interface Options {
+	/**
+	 * Enable or disable Chalk.
+	 *
+	 * @default true
+	 */
+	enabled?: boolean;
+
+	/**
+	 * Specify the color support for Chalk. By default,
+	 * color support is automatically detected based on
+	 * the environment.
+	 */
+	level?: Level;
+}
+
+export interface Constructor {
+	/**
+	 * Return a new Chalk instance.
+	 */
+	new (options?: Options): Chalk;
+
+	/**
+	 * Return a new Chalk instance.
+	 */
+	(options?: Options): Chalk;
+}
+
+/**
+ * Detect whether the terminal supports color.
+ */
+export interface ColorSupport {
+	/**
+	 * The color level used by Chalk.
+	 */
+	level: Level;
+
+	/**
+	 * Return whether Chalk supports basic 16 colors.
+	 */
+	hasBasic: boolean;
+
+	/**
+	 * Return whether Chalk supports ANSI 256 colors.
+	 */
+	has256: boolean;
+
+	/**
+	 * Return whether Chalk supports Truecolor 16 million colors.
+	 */
+	has16m: boolean;
+}
+
+export interface Chalk {
+	(...text: string[]): string;
+
+	(text: TemplateStringsArray, ...placeholders: string[]): string;
+
+	/**
+	 * Return a new Chalk instance.
+	 */
+	constructor: Constructor;
+
+	/**
+	 * Enable or disable Chalk.
+	 *
+	 * @default true
+	 */
+	enabled: boolean;
+
+	/**
+	 * The color support for Chalk. By default, color
+	 * support is automatically detected based on the
+	 * environment.
+	 */
+	level: Level;
+
+	/**
+	 * Use HEX value to set text color.
+	 *
+	 * @param color - Hexadecimal value representing the desired color.
+	 *
+	 * @example
+	 *
+	 * import chalk from 'chalk';
+	 *
+	 * chalk.hex('#DEADED');
+	 */
+	hex(color: string): this;
+
+	/**
+	 * Use keyword color value to set text color.
+	 *
+	 * @param color - Keyword value representing the desired color.
+	 *
+	 * @example
+	 *
+	 * import chalk from 'chalk';
+	 *
+	 * chalk.keyword('orange');
+	 */
+	keyword(color: string): this;
+
+	/**
+	 * Use RGB values to set text color.
+	 *
+	 * @param r - red
+	 * @param g - green
+	 * @param b - blue
+	 */
+	rgb(r: number, g: number, b: number): this;
```

Comment:
Use the full words, not `r` and `g`. Same with the other methods. The params then don't need any `@param` docs.